addons/binding/org.openhab.binding.s7/src/main/java/org/openhab/binding/s7/handler/S7BridgeHandler.java

```
@@ -0,0 +1,342 @@
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        destroyPollingJob();
+        destroyClient();
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing S7 bridge handler.");
+        createClient();
+
+        if (client != null && client.Connected) {
+            createPollingJob();
+        } else {
+            onConnectionLost();
+        }
+    }
+
+    private void createClient() {
+        if (getConfig().get(HOST) != null) {
+            if (client == null) {
+                client = new S7Client();
+                connectClient();
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to S7 Server. IP address not set.");
+        }
+    }
+
+    private boolean connectClient() {
+        if (client.Connected) {
+            return true;
+        }
+
+        String hostname = (String) getConfig().get(HOST);
+        int localTSAP = ((java.math.BigDecimal) getConfig().get(LOCAL_TSAP)).intValue();
+        int remoteTSAP = ((java.math.BigDecimal) getConfig().get(REMOTE_TSAP)).intValue();
+
+        client.SetConnectionParams(hostname, localTSAP, remoteTSAP);
+
+        client.Connect();
+
+        if (client.Connected) {
+            onConnectionEstablished();
+        }
+
+        return client.Connected;
+    }
+
+    private synchronized void onUpdate() {
+        destroyClient();
+        createClient();
+
+        destroyPollingJob();
+        createPollingJob();
+    }
+
+    private void destroyPollingJob() {
+        if (pollingJob != null) {
+            pollingJob.cancel(false);
+
+            while (!pollingJob.isDone()) {
+                try {
+                    Thread.sleep(5);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```

Do not use PrintStackTrace, in this case you can ignore this exception